text/0089-flexible-inflation.md

```diff
@@ -0,0 +1,260 @@
+
+> The naming used in this section is tentative, based on a WIP implementation, and subject to change
+> before finalization of this RFC.
+
+The new order splits the process for inflation into two steps:
+
+1. **Sourcing** the inflation amount: This step merely specifies by how much the chain intends to
+   inflation its token. This amount is not minted right away, and is instead passed over to the next
+   step for *distribution*.
+2. **Distributing** the aforementioned amount: A sequence of functions that decide what needs to be
+   done with the sourced inflation amount. This process is expected to _transfer_ the inflation
+   amount to any account that should receive it. This implies that the staking system should,
+   similar to treasury, have a key-less account that will act as a temporary pot for the inflation
+   amount.
+
+In very abstract terms, an example of the above process can be:
+
+* The chain inflates its token by a fixed 10% per year, an amount called `i`.
+* Payout 20% of `i` to the treasury account.
+* Payout 10% of what is left of `i` to the fellowship account.
+* Payout up to 70% of what is left of `i` to staking, depending on the staking rate.
+* Burn anything that is left.
+
+A proper configuration of this pallet should use `pallet_parameters` where possible to allow for any
+of the actual values used to specify `Sourcing` and `Distribution` to be changed via on-chain
+governance. Please see the [example configurations](#example-configurations) section for more
+details.
+
+In the new model, inflation can happen at any point in time. Since now a new pallet is dedicated to
+inflation, and it can internally store the timestamp of the last inflation point, and always inflate
+the correct amount. This means that while the duration of a staking era can is 1 day, the inflation
+process can happen eg. every hour. The opposite is also possible, although more complicated: The
+staking/treasury system can possibly receive their corresponding income on a weekly basis, while the
+era duration is still 1 day. That being said, we don't recommend using this flexibility as it brings
+no clear advantage, and is only extra complexity. We recommend the inflation to still happen shortly
+before the end of the staking era. This means that if the inflation `sourcing` or `distribution` is
+a function of the staking rate, it can reliably use the staking rate of the last era.
+
+Finally, as noted above, this RFC implies a new accounting system for staking to keep track of its
+staking reward. In short, the new process is as follows: `pallet_inflation` will mint the staking
+portion of inflation directly into a key-less account controlled by `pallet_staking`. At the end
+each era, `pallet_staking` will inspect this account, and move whatever amount is paid out into it
+to another key-less account associated withe era number. The actual payouts, initiated by stakers,
+will transfer from this era account into the corresponding stakers' account.
```

Comment:
What would happen if some era rewards go unclaimed (for more than 84 eras)?

Reply:
`pallet-staking` removes all era points of an era `n` once 84 eras have been passed. It should similarly burn any funds that are left in the pot account of era `n`. I believe the WIP implementation should already be doing that.